app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt

```diff
@@ -114,69 +136,113 @@ fun WorkoutCreationScreen(
-                                    exercises =
-                                        exerciseList.toMutableList() as MutableList<YogaExercise>))
-                          }
-                          WorkoutType.BODY_WEIGHT -> {
-                            workoutViewModel.addWorkout(
-                                BodyWeightWorkout(
-                                    workoutId = workoutViewModel.getNewUid(),
-                                    name = name,
-                                    description = description,
-                                    warmup = warmup,
-                                    exercises =
-                                        exerciseList.toMutableList()
-                                            as MutableList<BodyWeightExercise>))
-                          }
-                          else -> {}
+
+                  item {
+                    // Vertical line connecting the cards
+                    Column(
+                        horizontalAlignment = Alignment.CenterHorizontally,
+                        modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp)) {
+                          Box(
+                              modifier =
+                                  Modifier.size(8.dp)
+                                      .background(Color(0xFF9C7EEA), shape = CircleShape))
+                          Spacer(
+                              modifier =
+                                  Modifier.height(16.dp).width(2.dp).background(Color(0xFF9C7EEA)))
                         }
-                        Toast.makeText(context, "Workout successfully added", Toast.LENGTH_SHORT)
-                            .show()
-                        navigationActions.navigateTo(Screen.MAIN)
-                      },
-                      modifier = Modifier.padding(16.dp).testTag("saveButton")) {
-                        Text("Save")
-                      }
+                    Card(
+                        shape = RoundedCornerShape(16.dp),
+                        colors =
+                            CardDefaults.cardColors(
+                                containerColor = Color(0xFFD3D3D3)), // Couleur grise pour la carte
+                        modifier =
+                            Modifier.fillMaxWidth(0.9f)
+                                .padding(horizontal = 24.dp, vertical = 8.dp)) {
+                          // Box cliquable avec texte "+"
+                          Box(
+                              contentAlignment = Alignment.Center,
+                              modifier =
+                                  Modifier.fillMaxWidth()
+                                      .height(56.dp)
+                                      .clickable { showExerciseDialog = true }
+                                      .testTag("addExerciseButton")) {
+                                Text(text = "+", fontSize = 24.sp, color = Color.DarkGray)
```

Comment:
We should avoide strings defined like that. The preferable way is to define it in string.xml like that : 
<string name="addButton">+</string> and replace text = "+" by text = stringResource(id = R.string.addButton)